library/src/main/java/com/novoda/downloadmanager/lib/Criteria.java

```diff
@@ -0,0 +1,177 @@
+package com.novoda.downloadmanager.lib;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Criteria {
+
+    public static final Criteria ALL = new Criteria(null, null);
+
+    private final String selection;
+    private final String sort;
+    private final String[] selectionArguments;
+
+    Criteria(String selection, String sort, String... selectionArguments) {
+        this.selection = selection;
+        this.sort = sort;
+        this.selectionArguments = selectionArguments;
+    }
+
+    public String getSelection() {
+        return selection;
+    }
+
+    public String[] getSelectionArguments() {
+        return Arrays.copyOf(selectionArguments, selectionArguments.length);
+    }
+
+    public String getSort() {
+        return sort;
+    }
+
+    public static class Builder {
```

Comment:
Here you are using the `with...()` methods for appending values while in the `BatchQuery` the values are set/overwritten.

`new BatchQuery.Builder().withId(1).withId(2)` has a different meaning than `new Criteria.Builder().withSelection(A, wildcard).withArgument(1).withSelection(B, wildcard).withArgument(2)` 

better just use `selection()` and `argument()` similar to `or`, ..